backend/ohq/views.py

```diff
@@ -772,6 +777,103 @@ def list(self, request, *args, **kwargs):
 
         serializer = OccurrenceSerializer(occurrences, many=True)
         return JsonResponse(serializer.data, safe=False)
+    
+    def update(self, request, *args, **kwargs):
+        occurrence = self.get_object()
+        old_start = occurrence.start
+        old_end = occurrence.end
+        occurrence.start = datetime.strptime(request.data.get("start"), "%Y-%m-%dT%H:%M:%SZ").replace(
+            tzinfo=utc
+        )
+        occurrence.end = datetime.strptime(request.data.get("end"), "%Y-%m-%dT%H:%M:%SZ").replace(
+            tzinfo=utc
+        )
+        start_delta = occurrence.start - old_start
+        end_delta = occurrence.end - old_end
+        occurrence.save()
+
+        bookings = Booking.objects.filter(occurrence=occurrence).order_by("start")
+
+        for booking in bookings:
+            booking.start += start_delta
+            booking.end += end_delta
+            booking.save()
+
+        serializer = OccurrenceSerializer(occurrence)
+        return JsonResponse(serializer.data, safe=False)
 
     def get_queryset(self):
         return Occurrence.objects.filter(pk=self.kwargs["pk"])
+
+class BookingViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    retrieve:
+    Return a Booking.
+
+    list:
+    You should pass in an occurrence id, and all the bookings related to that occurrence will be returned to you.
+    Return a list of bookings.
+
+    create:
+    Create a booking.
+    occurrenceId is required in body.
+
+    update:
+    Update all fields in a Booking.
+    You must specify all of the fields or use a patch request.
+
+    partial_update:
+    Update certain fields in the Booking.
+    """
+
+    serializer_class = BookingSerializer
+    permission_classes = [BookingPermission | IsSuperuser]
+
+    def create(self, request, *args, **kwargs):        
```

Comment:
are we only allowed to create bookings in chronological order? For example, if an occurrence is from 1-3pm and no other bookings existed, I would only be able to book a slot at 1pm. If someone had booked starting at 1pm, I would only be able to book a 1:30pm spot (assuming interval is 30mins).